rust/parser/typeql.pest

```diff
@@ -277,12 +277,10 @@ undefine_struct = { STRUCT ~ identifier }
 
 // TYPE, LABEL AND IDENTIFIER CONSTRUCTS =======================================
 
-type_ref = { named_type | var }
-type_ref_optional = { type_ref ~ QUESTION }
+type_ref = { label_scoped | label | var }
 type_ref_list = { type_ref ~ SQ_BRACKET_OPEN ~ SQ_BRACKET_CLOSE }
-type_ref_any = { type_ref_optional | type_ref_list | type_ref }
 
-named_type = { value_type_primitive | label_scoped | label }
+named_type = { value_type_primitive | label }
```

Comment:
NamedType is used for arguments, return types & struct fields. Role types are not valid in the context.

Reply:
Maybe we need a better name than `NamedType`?
Maybe `InstantiableType`?

Reply:
Might be an idea, do you have any suggestions? If nothing pops up the current is okay, but not fantastic, i agree